contracts/contracts/sbtc-registry.clar

```diff
@@ -334,11 +371,35 @@
 )
 
 ;; Checks whether the contract-caller is a protocol contract
-(define-read-only (is-protocol-caller)
-  (validate-protocol-caller contract-caller)
+(define-read-only (is-protocol-caller (contract-flag (optional (buff 1))))
+  (validate-protocol-caller contract-flag contract-caller)
 )
 
 ;; Validate that a given principal is a protocol contract
-(define-read-only (validate-protocol-caller (caller principal))
-  (ok (asserts! (is-some (map-get? protocol-contracts caller)) ERR_UNAUTHORIZED))
+(define-read-only (validate-protocol-caller (contract-flag (optional (buff 1))) (caller principal))
```

Comment:
It doesn't appear that there's any usage of calling this with `none` is there? This also seems kinda dangerous - I see that the purpose is, if you pass `none`, it just says "is this caller any of the roles". But IMO that shouldn't ever be used for authorization.

Reply:
Ah, I see it's used in `protocol-transfer`.

We could solve this with a two "inverse" maps. So you'd have like:

```clarity
;; This is for role -> contract
(define-map protocol-contracts (buff 1) principal)
;; This is for contract -> role
(define-map protocol-contract-role principal (buff 1)
```

Then, in things like `protocol-transfer`, you could just check that `map-get? protocol-contract-role contract-caller` _has_ a role